### PR TITLE
Makisheva/files preview

### PR DESCRIPTION
**Pull request by AlbinaMakisheva**

### CODE REVIEW

**Comment by mirekys** [comment]
Trigger "button" doesn't look & feel like an actual button...in DOM, its just plain icon

Screenshot 2024-05-23 at 16 08 59

**Comment by mirekys** [comment]
Incorrect preview url used by  preview modal
Screenshot 2024-05-23 at 16 11 05

**Comment by mirekys** [comment]
Iframe close button should be at top right corner where everyone expects it (not left bottom)

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import $ from "jquery";
+
+$("#preview-modal").modal();
+const iframe = document.querySelector(".fileIframe");
+
+function openModal(event) {
+  const fileTitle = event.target.dataset.title;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js:7]: Whole preview url should come from data attribute (& from file links) and should not be concatenated like this

**Review by mirekys — CHANGES_REQUESTED** [review]

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js (hunk 1 above)
Comment (AlbinaMakisheva) [reply oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js:7]: There is no added preview url https://github.com/Narodni-repozitar/nr-docs/blob/b3e4ad113ec8bd79648c6e0b254e1993c75598c7/documents/services/files/config.py#L41

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.variables [hunk 2]
```diff
@@ -0,0 +1 @@
+@openPreviewIconOnHoverColor: gray;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.variables:1]: Why do we need special variable just for this?

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides [hunk 3]
```diff
@@ -30,4 +30,11 @@
       display: none !important;
     }
   }
+  
 }
+.rel-mb-1{
+    & .openPreviewIcon:hover{
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides:36]: Why do we need site-specific special css rules just for usual button behaviour

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/definitions/preview-modal.less [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+@import (multiple) "../../theme.config";
+
+/*
+  This is the styling of your component. The class should start with .ui.<component-name>
+
+  Note: use variables here, not the exact values. Placing the values to another file (sample.variables)
+  will allow users to override these values.
+ */
+.ui.modal#preview-modal {
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/definitions/preview-modal.less:36]: Why do we need special css rules just for this, isn't the usual SUI Modal already rendered like this? Isn't there any SUI prop to reach desired ui? This modal should look consistenly, just like any other modals used in our UI

File: oarepo_ui/templates/components/datafields/files/FilesViewer.jinja [hunk 5]
```diff
@@ -1,30 +1,39 @@
...
+  </thead>
+  <tbody>
+    {% for file in array(files.entries) %}
+    <tr>
+      <td>
+        <a href="{{ file.links.content }}">{{ file.key }}</a>
+      </td>
+      <td>{{ file.size | filesizeformat }}</td>
+      <td>
+        <i class="ui icon zoom openPreviewIcon" data-link="{{ file.links.preview }}"></i>
```
Comment (mirekys) [inline oarepo_ui/templates/components/datafields/files/FilesViewer.jinja:24]: Still just a plain icon, no actionable button

**Review by mirekys — CHANGES_REQUESTED** [review]

File: oarepo_ui/templates/components/datafields/files/FilesViewer.jinja [hunk 6]
```diff
@@ -1,30 +1,44 @@
...
+    <tr>
+      <td>
+        <a href="{{ file.links.content }}">{{ file.key }}</a>
+      </td>
+      <td>{{ file.size | filesizeformat }}</td>
+      <td>
+        <button class="ui button">
+          <i
+            class="ui icon zoom openPreviewIcon"
+            data-link="{{ file.links.preview }}"
```
Comment (mirekys) [inline oarepo_ui/templates/components/datafields/files/FilesViewer.jinja:27]: link -> preview-link

**Review by mirekys — CHANGES_REQUESTED** [review]

File: oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import $ from "jquery";
+
+const iframe = document.querySelector(".fileIframe");
+
+$("#preview-modal").modal({
+  onShow: function () {
+    const contentHeight = window.innerHeight * 0.8;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js:7]: This is not responsive, once the modal is shown, it is never resized according to window size changes, possibly overflowing content

Comment (mirekys) [reply oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/components/filepreview.js:7]: https://www.w3schools.com/howto/howto_css_responsive_iframes.asp

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides [hunk 8]
```diff
@@ -1,33 +1,34 @@
 :not(.ui.grid).only {
   &.mobile:not(.tablet) {
     @media all and (min-width: @tabletBreakpoint) {
-      display: none !important;
+      display: none !important                        ;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides:4]: extra whitespace

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides [hunk 9]
```diff
@@ -1,33 +1,34 @@
...
     }
   }
 
   &.tablet {
     &.mobile {
       @media all and (min-width: @computerBreakpoint) {
-        display: none !important;
+        display: none !important                          ;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides:11]: extra whitespace

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides [hunk 10]
```diff
@@ -1,33 +1,34 @@
...
       }
     }
 
     &.computer {
       @media all and (max-width: @largestMobileScreen) {
-        display: none !important;
+        display: none !important                           ;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides:17]: extra whitespace

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides [hunk 11]
```diff
@@ -1,33 +1,34 @@
...
       }
     }
 
     &:not(.computer):not(.mobile) {
       @media not all and (min-width: @tabletBreakpoint) and (max-width: @largestTabletScreen) {
-        display: none !important;
+        display: none !important                                                                  ;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides:23]: extra whitespace

File: oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides [hunk 12]
```diff
@@ -1,33 +1,34 @@
...
       }
     }
   }
 
   &.computer:not(.tablet) {
     @media all and (max-width: @largestTabletScreen) {
-      display: none !important;
+      display: none !important                           ;
```
Comment (mirekys) [inline oarepo_ui/theme/assets/semantic-ui/less/oarepo_ui/themes/default/globals/site.overrides:30]: extra whitespace

**Review by mirekys — CHANGES_REQUESTED** [review]

**Comment by mirekys** [comment]
Hide preview button when preview is not possible (no preview link generated for unsupported file type)

**Review by mirekys — CHANGES_REQUESTED** [review]
Hide preview button when preview is not possible (no preview link generated for unsupported file type)

File: oarepo_ui/templates/components/datafields/files/FilesViewer.jinja [hunk 13]
```diff
@@ -1,30 +1,47 @@
...
+          ></a>
+        </button>
+      </td>
+    </tr>
+    {% endfor %}
+  </tbody>
 </table>
+
+<div class="ui modal fullscreen" id="preview-modal">
+  <i class="close icon"></i>
```
Comment (mirekys) [inline oarepo_ui/templates/components/datafields/files/FilesViewer.jinja:44]: Same semantic problem as with preview button, this has to be a button, not just some icon

**Review by mirekys — CHANGES_REQUESTED** [review]

**Review by mirekys — DISMISSED** [review]

**Review by mirekys — APPROVED** [review]

**Pull request merged**